core/meta/inc/TGenericClassInfo.h

```diff
@@ -12,19 +12,121 @@
 #ifndef ROOT_TGenericClassInfo
 #define ROOT_TGenericClassInfo
 
```

Comment:
```suggestion
#ifndef ROOT_Rtypes
// Include RTypes.h outside of the code guard to insure the intended
// ordering of RTypes.h and TGenericClassInfo.h
#include "Rtypes.h"
#endif

#ifndef ROOT_TGenericClassInfo
#define ROOT_TGenericClassInfo
```
This also solves the problem while avoiding the need to migrate the type declaration from one file to the other.